pkg/solana/logpoller/loader.go

```diff
@@ -90,29 +90,54 @@ func (c *EncodedLogCollector) BackfillForAddress(ctx context.Context, address st
 		return err
 	}
 
-	sigs, err := c.client.GetSignaturesForAddressWithOpts(ctx, pubKey, &rpc.GetSignaturesForAddressOpts{
-		Commitment:     rpc.CommitmentFinalized,
-		MinContextSlot: &fromSlot,
-	})
-	if err != nil {
-		return err
-	}
+	var (
+		lowestSlotRead uint64
+		lowestSlotSig  solana.Signature
+	)
 
-	slots := make(map[uint64]struct{})
+	for lowestSlotRead > fromSlot {
```

Comment:
I think it's important we start from `fromSot` and read upward from there rather than starting at the current block and reading down to fromSlot. Doing it backwards will maximize latency because LogPoller won't be able to start saving the first slot (fromSlot) until all of them are read.  I know we're not guaranteeing the order at this level, but I think it would be a shame to have the typical case be maximally inefficient and correct order being maximally unlikely.

Reply:
I left a similar comment in the JIRA about this last week, although that was before you convinced me we didn't need to guarantee the order here.

Reply:
Hmm... yeah I guess that does pose a slight problem.

I don't know of any case where we'd care about having an upper limit, so that part is fine.  But the lower limit might be tricky if you can get back some signatures from a slot but not others. (For example, if 1000 is the maximum # it can return... then what if txs 998 - 1002 are all in a block together? I guess if it only returns 998, 999 and 1000 then we'd have to set the next lower limit as the same block, and get back a few that we already know about.  I think that's probably fine, since it's unlikely a single block would have more than 1 transaction we care about anyway.